### ISSUE TITLE
Key error on add_contig_taxonomy.py

### ISSUE THREAD

**Issue by SithWijesinghe** [KwanLab/Autometa#98]
### User checklist

- [ no. Version 1.0 ] Are you using the latest release?
- [ no 2.7.15 ] Are you using python 3?
- [ yes ] Did you check previous issues to see if this has already been mentioned?
- [ Linux ] Are you using a Mac or Linux machine?

#### Description

I've used the following workflow successfully on all but one of my datasets after issue #94 .

```
calculate_read_coverage.py -p $THREADS -a $ASSEMBLY -F $READ1 -R $READ2 -o ${OUTDIR}/coverage

make_taxonomy_table.py --assembly $ASSEMBLY -p $THREADS \
--db_dir $AUTOMETA_DB -v ${OUTDIR}/coverage/coverage.tab \
-o ${OUTDIR}/taxtable --length_cutoff $CONTIG_LEN

run_autometa.py -p $THREADS -o $OUTDIR --length_cutoff $CONTIG_LEN \
--db_dir $AUTOMETA_DB -v ${OUTDIR}/coverage/coverage.tab \
-t ${OUTDIR}/taxtable/taxonomy.tab --assembly $ASSEMBLY

ML_recruitment.py --contig_tab ${OUTDIR}/recursive_dbscan_output.tab \
--k_mer_matrix ${OUTDIR}/k-mer_matrix --out_table ${OUTDIR}/ML_recruitment_output.tab

cluster_process.py --bin_table ${OUTDIR}/ML_recruitment_output.tab --column \
ML_expanded_clustering --fasta ${OUTDIR}/taxtable/Bacteria.fasta --do_taxonomy \
--db_dir $AUTOMETA_DB --output_dir ${OUTDIR}/cluster_process
```
In one of the samples, running ```make_taxonomy_table.py``` results in a key error when running ```add_contig_taxonomy.py```, and make_taxonomy_table step exits without generating a tax table.

I suspect this might be a sample specific issue since 15 of my other samples did not have this issue.

Any advice on how to deal with this? Thank you!

#### System Environment

- Operating System: Ubuntu 18.04.4 LTS
- RAM: 32000M
- Disk: It's an HPC. I don't have the disc capacity, I just know there's sufficient storage. sorry.

<details><summary>Log/Error information generated by Autometa.</summary><p>

```
...........................................
Loading query sequences...  [0.049s]
Closing the input file...  [0s]
Closing the output file...  [0.413s]
Closing the database file...  [0s]
Deallocating taxonomy...  [0s]
Total time = 12544.1s
Reported 13697266 pairwise alignments, 13700498 HSPs.
77586 queries aligned.
20:15:25: Beginning LCA
00:00:15: Finished building tree
00:01:53: Finished constructing sparse table
00:02:04: Finished extracting blastp orfs. parsing prot.acc2taxid DB
00:16:44: Finished acc2taxid translation dict
00:16:49: Finished acc2taxid conversion
00:17:33: Finished LCA query
100%|██████████| 3225612/3225612 [00:05<00:00, 594359.58it/s]
100%|█████████▉| 2255422/2255423 [00:06<00:00, 337654.76it/s]
 26%|██▋       | 20478/77586 [00:00<00:01, 47931.91it/s]
2020-07-03 06:33:28 Processing taxid names
2020-07-03 06:33:34 Processing taxid nodes
2020-07-03 06:33:41 Parsing lca taxonomy table
Traceback (most recent call last):
  File "/mnt/transient_nfs/conda/envs/autometa/autometa/pipeline/add_contig_taxonomy.py", line 351, in <module>
    classifications = parse_lca(tax_table_path)
  File "/mnt/transient_nfs/conda/envs/autometa/autometa/pipeline/add_contig_taxonomy.py", line 204, in parse_lca
    taxid = nodes[taxid]['parent']
KeyError: 211759
lca_functions up-to-date
make_taxonomy_table.py, run_command: /mnt/transient_nfs/conda/envs/autometa/autometa/pipeline/fasta_length_trim.pl /scratch/s4187725/marsupial/pilot/output/bin/1_assembly/megahit/206_Yellow_bellied_glider/final.contigs.fa 1500 /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.fa
Prodigal output not found. Running prodigal...
make_taxonomy_table.py, run_command: prodigal -i /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.fa -a /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.orfs.faa -p meta -m -o /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.txt
Could not find /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.orfs.blastp. Running diamond blast...
make_taxonomy_table.py, run_command: diamond blastp --query /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.orfs.faa --db /scratch/s4187725/marsupial/pilot/data/autometa_db_new/nr.dmnd       --evalue 1e-5 --max-target-seqs 200 -p 16 --outfmt 6 --out /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.orfs.blastp -t /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/tmp
Could not find /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.orfs.lca. Running lca...
make_taxonomy_table.py, run_command: /mnt/transient_nfs/conda/envs/autometa/autometa/pipeline/lca.py database_directory /scratch/s4187725/marsupial/pilot/data/autometa_db_new /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.orfs.blastp
Running add_contig_taxonomy.py...
make_taxonomy_table.py, run_command: /mnt/transient_nfs/conda/envs/autometa/autometa/pipeline/make_contig_table.py -a /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.fa -o /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.tab -c /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/coverage/coverage.tab
make_taxonomy_table.py, run_command: /mnt/transient_nfs/conda/envs/autometa/autometa/pipeline/add_contig_taxonomy.py /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.tab /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.orfs.lca /scratch/s4187725/marsupial/pilot/data/autometa_db_new /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/taxonomy.tab
make_taxonomy_table.py: Error, the command:
/mnt/transient_nfs/conda/envs/autometa/autometa/pipeline/add_contig_taxonomy.py /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.tab /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/final.contigs.filtered.orfs.lca /scratch/s4187725/marsupial/pilot/data/autometa_db_new /scratch/s4187725/marsupial/pilot/output/bin/3_autometa/206_Yellow_bellied_glider/taxtable/taxonomy.tab
failed, with exit code 1
```

</p></details>

**Comment by jason-c-kwan** [comment]
@WiscEvan will probably chime in here, but it might be helpful if we could get hold of the files that are used in the last command, so `final.contigs.filtered.tab`, `final.contigs.filtered.orfs.lca` and `taxonomy.tab`, so that we can reproduce the error. Only if you are comfortable with sharing, of course.

**Comment by SithWijesinghe** [comment]
Sure, no problem. I have attached ```final.contigs.filtered.tab``` and ```final.contigs.filtered.orfs.lca```. 

A ```taxonomy.tab``` file was not generated.

[final.contigs.filtered.zip](https://github.com/KwanLab/Autometa/files/4871971/final.contigs.filtered.zip)

**Comment by WiscEvan** [comment]
Hello, I have published a more recent release that should fix this issue. The issue is a result of the NCBI databases being out of sync. The taxid that resulting in a `KeyError` has been deprecated/suppressed by NCBI. This will now be updated from merged.dmp to the most recent taxid. With the most recent release. merged.dmp from taxdump.tar.gz will also be extracted and processed. Can you give this a try with [release 1.0.1](https://github.com/KwanLab/Autometa/releases/tag/1.0.1) and let us know if you are still encountering this error?

**Comment by WiscEvan** [comment]
@SithWijesinghe were you able to resolve this issue with the 1.0.1 release?

**Comment by SithWijesinghe** [comment]
@WiscEvan it's resolved with v 1.0.1 now, thank you!
